cloudpathlib/cloudpath.py

```diff
@@ -609,6 +611,90 @@ def rmtree(self):
             )
         self.client._remove(self)
 
+    def upload_from(
+        self, source: Union[str, os.PathLike], force_overwrite_to_cloud: bool = False
```

Comment:
It's not clear to me why we need `force_overwrite_to_cloud` if `source` can be an arbitrary file. I would expect this method to just always overwrite. Conceptually similar methods for moving or copying will just overwrite things. 

Relatedly, I think the checking logic for this should remain in `_upload_local_to_cloud` instead of the more general `_upload_file_to_cloud`.

Reply:
Yeah, I guess that overwriting your canonical data in the cloud is potentially the worst thing you could do so we were being extra careful to be explicit here that you want to overwrite. I'm on the fence for this one as a default.